### PR TITLE
Debug route adjustments

### PR DESCRIPTION
**Pull request by crookedneighbor**
This is an idea for a way to set up controller level middlware once, instead of for each route.

@paglias @TheHollidayInn @sabrecat

### CODE REVIEW

**Comment by crookedneighbor** [comment]
Not sure how we would API doc this, or if we even need to since it is not a proper route.

File: website/src/controllers/api-v3/development.js [hunk 1]
```diff
@@ -4,6 +4,12 @@ import checkForDevelopmentMode from '../../middlewares/api-v3/developmentMode';
 
 let api = {};
 
+api.development = {
```
Comment (crookedneighbor) [inline website/src/controllers/api-v3/development.js:7]: Maybe `api.all` is a better name?

**Comment by paglias** [comment]
Looks good to me, we could also do like this:

`_.each(api, (route) => {
  route.middleware = [checkForDev, auth, cron]
})`

but it's the same for me

**Comment by crookedneighbor** [comment]
@paglias The downside to the `_.each` solution is that:

1) You'd have to declare it at the bottom of the file, which I think is weird considering they are basically the pre-requisites for these routes
2) You couldn't just overwrite the middleware, you'd have to add on to the middleware array in case some specific route needs more middleware. In fact, you'd probably have to set them at the _beginning_ of the middleware array to make sure they were checked first.

The downside with the solution I'm proposing in this PR is that it does not work for the base route. IE, if we declared this on the groups controller as `groups/*` it wouldn't catch `POST /groups` or `GET /groups`. We could use a regex instead:

``` js
url: /groups(\/.*)?/
```

But that seems a little heavy handed to me.

**Comment by paglias** [comment]
Good points, feel free to merge!

File: website/src/controllers/api-v3/development.js (hunk 1 above)
Comment (TheHollidayInn) [inline website/src/controllers/api-v3/development.js:7]: I think naming this as the namespace is good. So, if we change this to debug.js then I think this should be called api.debug.

**Comment by TheHollidayInn** [comment]
This looks great. I like the abstraction.

**Pull request merged**